File: Python/Antylopa.py
from Zwierze import Zwierze
from Organizm import Organizm

class Antylopa(Zwierze):
    def __init__(self, x, y, dz, swiat):
        super().__init__(x, y, dz, 4, 4, swiat.IDANTYLOPA, swiat)

    def rysowanie(self):
        return "#a6730c"

    def nazwa(self):
        return "Antylopa"

    def kolizja(self, organizm, kierunek):
        if self.__class__ == organizm.__class__:
            super().kolizja(organizm, kierunek)
        else:
            random = self.swiat.rand.randint(0, 2)
            if random == 0:
                if organizm.GetY() != 0 and self.swiat.GetOrganizmy(organizm.GetX(), organizm.GetY() - 1) == None:
                    self.swiat.dodajWydarzenie(self, organizm, self.swiat.UCIEKA)
                    self.idzXY(organizm.GetX(), organizm.GetY() - 1)
                elif organizm.GetY() != self.swiat.Getwyskosc() - 1 and self.swiat.GetOrganizmy(organizm.GetX(), organizm.GetY() + 1) == None:
                    self.swiat.dodajWydarzenie(self, organizm, self.swiat.UCIEKA)
                    self.idzXY(organizm.GetX(), organizm.GetY() + 1)
                elif organizm.GetX() != 0 and self.swiat.GetOrganizmy(organizm.GetX() - 1, organizm.GetY()) == None:
                    self.swiat.dodajWydarzenie(self, organizm, self.swiat.UCIEKA)
                    self.idzXY(organizm.GetX() - 1, organizm.GetY())
                elif organizm.GetX() != self.swiat.Getszerokosc() - 1 and self.swiat.GetOrganizmy(organizm.GetX() + 1, organizm.GetY()) == None:
                    self.swiat.dodajWydarzenie(self, organizm, self.swiat.UCIEKA)
                    self.idzXY(organizm.GetX() + 1, organizm.GetY())
                elif self.swiat.Getuklad_planszy() == 6:
                    if self.GetX() % 2 == 0:
                        if organizm.GetX() != self.swiat.Getszerokosc() - 1 and organizm.GetY() != 0 and self.swiat.GetOrganizmy(organizm.GetX() + 1, organizm.GetY() - 1) == None:
                            self.swiat.dodajWydarzenie(self, organizm, self.swiat.UCIEKA)
                            self.idzXY(organizm.GetX() + 1, organizm.GetY() - 1)
                        elif organizm.GetX() != 0 and organizm.GetY() != 0 and self.swiat.GetOrganizmy(organizm.GetX() - 1, organizm.GetY() - 1) == None:
                            self.swiat.dodajWydarzenie(self, organizm, self.swiat.UCIEKA)
                            self.idzXY(organizm.GetX() - 1, organizm.GetY() - 1)
                    else:
                        if organizm.GetX() != self.swiat.Getszerokosc() - 1 and organizm.GetY() != self.swiat.Getwyskosc() - 1 and self.swiat.GetOrganizmy(organizm.GetX() + 1, organizm.GetY() + 1) == None:
                            self.swiat.dodajWydarzenie(self, organizm, self.swiat.UCIEKA)
                            self.idzXY(organizm.GetX() + 1, organizm.GetY() + 1)
                        elif organizm.GetX() != 0 and organizm.GetY() != self.swiat.Getwyskosc() - 1 and self.swiat.GetOrganizmy(organizm.GetX() - 1, organizm.GetY() + 1) == None:
                            self.swiat.dodajWydarzenie(self, organizm, self.swiat.UCIEKA)
                            self.idzXY(organizm.GetX() - 1, organizm.GetY() + 1)
            else:
                super().kolizja(organizm, kierunek)

    def stworz_nowe_zwierze(self, x, y, sw):
        org = Antylopa(x, y, True, sw)

    def ileRuchu(self):
        return 2

    def CzyUcieka(self):
        random = self.swiat.rand.randint(0, 2)
        if random == 0:
            return True
        return False

File: C++/Antylopa.cpp
#include <iostream>
#include <windows.h>
#include "Antylopa.h"

Antylopa::Antylopa(int x, int y, Swiat* swiat) : Zwierze(x, y, 4, 4, IDANTYLOPA, swiat)
{
}

Antylopa::Antylopa(int x, int y, bool dz, Swiat* swiat) : Zwierze(x, y, dz, 4, 4, IDANTYLOPA, swiat)
{
}

void Antylopa::rysowanie()
{
	HANDLE col;
	col = GetStdHandle(STD_OUTPUT_HANDLE);
	SetConsoleTextAttribute(col, KOLORANTYLOPA);
	std::cout << "A";
}

std::string Antylopa::nazwa()
{
	return "A";
}

void Antylopa::kolizja(Organizm* organizm, int kierunek)
{
	if (typeid(*this) == typeid(*organizm))
	{
		Zwierze::kolizja(organizm, kierunek);
	}
	else
	{
		int random = rand() % 2;
		if (random == 0)
		{
			if (organizm->GetY() != 0 && this->swiat->GetOrganizmy(organizm->GetX(), organizm->GetY() - 1) == nullptr)
			{
				this->swiat->dodajWydarzenie(this, organizm, UCIEKA);
				this->idzXY(organizm->GetX(), organizm->GetY() - 1);
			}
			else if (organizm->GetY() != this->swiat->Getwyskosc() - 1 && this->swiat->GetOrganizmy(organizm->GetX(), organizm->GetY() + 1) == nullptr)
			{
				this->swiat->dodajWydarzenie(this, organizm, UCIEKA);
				this->idzXY(organizm->GetX(), organizm->GetY() + 1);
			}
			else if (organizm->GetX() != 0 && this->swiat->GetOrganizmy(organizm->GetX() - 1, organizm->GetY()) == nullptr)
			{
				this->swiat->dodajWydarzenie(this, organizm, UCIEKA);
				this->idzXY(organizm->GetX() - 1, organizm->GetY());
			}
			else if (organizm->GetX() != this->swiat->Getszerokosc() - 1 && this->swiat->GetOrganizmy(organizm->GetX() + 1, organizm->GetY()) == nullptr)
			{
				this->swiat->dodajWydarzenie(this, organizm, UCIEKA);
				this->idzXY(organizm->GetX() + 1, organizm->GetY());
			}
		}
		else
		{
			Zwierze::kolizja(organizm, kierunek);
		}
	}
}

void Antylopa::stworz_nowe_zwierze(int x, int y, Swiat* sw)
{
	Organizm* org = new Antylopa(x, y, true, sw);
}

int Antylopa::ileRuchu()
{
	return 2;
}

bool Antylopa::CzyUcieka()
{
	int random = rand() % 2;
	if (random == 0)
		return true;
	return false;
}

Antylopa::~Antylopa()
{
}